### PR TITLE
new algo

### PR DESCRIPTION
**Pull request by jjaskirat-ssingh**
## **Description**
- Introduced a new Restricted Boltzmann Machine (RBM) model in PyTorch with forward pass, Gibbs sampling, and sampling functionalities.
- Added training and evaluation routines for the RBM, including a method to visualize generated samples.
- Created a requirements file to manage project dependencies.

___

## **Changes walkthrough**
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>main.py</strong><dd><code>Implementation of Restricted Boltzmann Machine in PyTorch</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

rbm/main.py
<li>Implemented a Restricted Boltzmann Machine (RBM) model class with <br>methods for forward pass, Gibbs sampling, and sampling from the model.<br> <li> Added a training function that includes a naive approach for <br>initializing from a random distribution every gradient step.<br> <li> Included an evaluation function based on reconstruction error and a <br>method for displaying generated samples as images.<br> <li> Created a main function to parse arguments and run the training, <br>evaluation, and image generation processes.

</details>
    

  </td>
  <td><a href="https://github.com/experiment-org-jaskirat/pytorch-examples/pull/1/files#diff-1ee8bc34aee40a71e3feff3970f884a12fb864290ed58f46ece889c5e34be1ab">+147/-0</a>&nbsp; </td>
</tr>                    
</table></td></tr><tr><td><strong>Configuration changes</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>requirements.txt</strong><dd><code>Requirements for RBM Project</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

rbm/requirements.txt
<li>Specified dependencies for the RBM project including torch, <br>torchvision, and matplotlib.

</details>
    

  </td>
  <td><a href="https://github.com/experiment-org-jaskirat/pytorch-examples/pull/1/files#diff-3c24c43247430191dfc5a328e43a24003fed56f60a5347f56cb0192d1d20324b">+3/-0</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    
</table></td></tr></tr></tbody></table><hr>

<details><summary><strong>🔍 Anti-patterns Detected:</strong></summary><hr>
<details><summary><strong>rbm/main.py</strong></summary><div><table><tr><th>Issue</th><th>Lines</th></tr><tr><td>If possible, it is better to rely on automatic pinning in PyTorch to avoid undefined behavior and for efficiency</td><td><a href='https://github.com/experiment-org-jaskirat/pytorch-examples/blob/7363507b36778d3daa9a945fd2573ef5fa850d59/rbm/main.py#L129:L129' target='_blank'>129-129</a></td></tr></table></div></details></details><details>
<summary><strong>💡 Usage Guide</strong></summary>

### Checking Your Pull Request
Every time you make a pull request, our system automatically looks through it. We check for security issues, mistakes in how you're setting up your infrastructure, and common code problems. We do this to make sure your changes are solid and won't cause any trouble later.

### Talking to CodeAnt AI
Got a question or need a hand with something in your pull request? You can easily get in touch with CodeAnt AI right here. Just type the following in a comment on your pull request, and replace "Your question here" with whatever you want to ask:
<pre>
<code>@codeant-ai ask: Your question here</code>
</pre>
This lets you have a chat with CodeAnt AI about your pull request, making it easier to understand and improve your code.

### Retrigger review
Ask CodeAnt AI to review the PR again, by typing:
<pre>
<code>@codeant-ai: review</code>
</pre>

### Check Your Repository Health
To analyze the health of your code repository, visit our dashboard at [app.codeant.ai](https://app.codeant.ai). This tool helps you identify potential issues and areas for improvement in your codebase, ensuring your repository maintains high standards of code health.

</details>

### CODE REVIEW

**Comment by jjaskirat-ssingh** [comment]
@codeant-ai ask: Can you provide the correct implementation of Random Initialization in Training Loop.

**Comment by jjaskirat-ssingh** [comment]
@CodeAnt-AI ask: Also, can you suggest provide the code for the suggested approach for Binary Data Transformation?
